protocols/shell-workspace-creation-v1.xml

```diff
@@ -0,0 +1,321 @@
+    in this protocol or compositor-specific companion protocols.
+
+    A client may attempt to discover potential creation methods by
+    creating new workspace targets, checking their capabilities and
+    destroying them without use.
+    
+    The key words "must", "must not", "required", "shall", "shall not",
+    "should", "should not", "recommended",  "may", and "optional" in this
+    document are to be interpreted as described in IETF RFC 2119.
+
+    Warning! The protocol described in this file is intended as a stopgap
+    and is expected to be superseded by a solution in wayland-protocols.
+    Clients should not assume this protocol will continue to exist in the
+    future.
+  </description>
+
+  <interface name="shell_workspace_creation_target_descriptor_v1" version="1">
+    <description summary="opaque workspace creation target descriptor">
+      The workspace creation target descriptor is an opaque object
+      that describes where and how a workspace will be created.
+
+      Because shell_workspace_creation_target_descriptor_v1 objects
+      are created from multiple independent factory interfaces,
+      the shell_workspace_creation_target_descriptor_v1 interface
+      is frozen at version 1.
+    </description>
+
+    <request name="destroy" type="destructor">
+      <description summary="destroy the descriptor">
+        Destroys the object.
+      </description>
+    </request>
+  </interface>
+
+  <interface name="shell_workspace_target_creation_manager_v1" version="1">
+    <request name="destroy" type="destructor">
+      <description summary="destroy the manager">
+        Destroy the manager. Destroying the manager does not destroy objects
+        created by the manager.
+      </description>
+    </request>
+
+    <request name="get_target">
+      <description>
+        This request creates a new workspace target from a given workspace
+        target descriptor.
+
+        The descriptor may be destroyed following this request.
+
+        Because the target descriptor cannot be versioned, this request
+        is used to create a versionable target from the descriptor.
+      </description>
+
+      <arg name="target" type="new_id" interface="shell_workspace_creation_target_v1"/>
+      <arg name="descriptor" type="new_id" interface="shell_workspace_creation_target_descriptor_v1"/>
+    </request>
+  </interface>
+
+  <interface name="shell_workspace_arbitrary_creation_manager_v1" version="1">
+    <request name="destroy" type="destructor">
+      <description summary="destroy the manager">
+        Destroy the manager. Destroying the manager does not destroy objects
+        created by the manager.
+      </description>
+    </request>
+
+    <request name="get_arbitrary_target">
+      <description summary="create a workspace target relative to another workspace">
+        This requests creates a workspace target that targets an arbitrary
+        compositor chosen name and location for the workspace.
```

Comment:
So this is like "yeah i just want a new workspace put it wherever"?

Reply:
Yeah. In most systems it would go at the end, but thats compositor policy.